webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TypeAheadSelect } from 'patternfly-react';
+import classNames from 'classnames';
+import { bindMethods, debounceMethods, noop } from '../../common/helpers';
+import AutoCompleteMenu from './components/AutoCompleteMenu';
+import AutoCompleteSearchButton from './components/AutoCompleteSearchButton';
+import AutoCompleteError from './components/AutoCompleteError';
+import AutoCompleteAux from './components/AutoCompleteAux';
+import AutoCompleteFocusShortcut from './components/AutoCompleteFocusShortcut';
+import { STATUS } from '../../constants';
+import { TRIGGERS, KEYCODES } from './AutoCompleteConstants';
+import { translate as __ } from '../../common/I18n';
+import './auto-complete.scss';
+
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    bindMethods(this, [
+      'handleClear',
+      'handleInputChange',
+      'handleResultsChange',
+      'handleInputFocus',
+      'getResults',
+      'windowKeyPressHandler',
+      'disableHTMLAutocomplete',
+      'handleKeyDown',
+    ]);
+    this._typeahead = React.createRef();
+    debounceMethods(this, 500, ['handleLoading']);
+  }
+
+  componentDidMount() {
+    window.addEventListener('keypress', this.windowKeyPressHandler);
```

Comment:
any thoughts about [this package](https://github.com/oliviertassinari/react-event-listener), a more declarative way for global events:
```js
 <EventListener target={document} onMouseMoveCapture={this.handleMouseMove} />
```
doesn't need to be in this PR scope

Reply:
overkill? can you explain why?
I think it much cleaner then using  `window` directly from react

Reply:
I think it takes much longer to write the event listener,
you also need to wrap your component with the `EventListener`.
in my opinion the render should return the least amount of components
and the most relevant to its scope, the logic and event handling should live inside
React life cycle methods.

Another thought is that the render can trigger this component again and again,
adding more event listeners than needed unlike setting a listener once
in `componentDidMount` and removing it in `componentWillUnmount`

Reply:
Adding an event listener on `componentDidMount` will increase browser event list as many times as you reuse that component. you can read more about it [here](https://hackernoon.com/do-you-still-register-window-event-listeners-in-each-component-react-in-example-31a4b1f6f1c8) 

as I mentioned before, I don't think this should block this PR. just a thought to share.

Reply:
although you are removing it on `willUnmount`? still need to investigate it..
what if you need 2 event listeners or more ? you'll need 2 wrappers ?
anyway thanks for sharing :+1:

Reply:
after discussing it more offline, I agree that creating events-registration mechanism is very important for us, not sure if it will be easier with the package you've mentioned or without.. do you know if it works with such mechanism?